tests/test_auth.py

```diff
@@ -69,6 +69,91 @@ def test_signout(self):
         response = self.client.post('/api/v1/users/auth/signout/')
         self.assertEqual(response.status_code, 200)
 
+    def test_username_taken(self):
+        """ Test if the username is already taken """
+        self.client.post(
+            '/api/v1/users/auth/signup/',
+            data=json.dumps(self.data["user"]),
+            content_type=("application/json")
+        )
+
+        response = self.client.post(
+            '/api/v1/users/auth/signup/',
+            data=json.dumps(self.data["user"]),
+            content_type=("application/json")
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_email_taken(self):
+        """ Test if the email is already taken """
+        self.client.post(
```

Comment:
for "DRY-er" code you could define the register as a function that you call whenever you require to make a post to `/api/v1/users/auth/signup/` ... unless in cases where you need to test like passwords not matching.